.github/workflows/docs.yaml

```diff
@@ -0,0 +1,55 @@
+name: Docs Preview
+
+on:
+  pull_request:
+
+jobs:
+  preview_docs:
+    timeout-minutes: 30
+    name: Docs preview
+    if: "github.event_name == 'pull_request'"
+    runs-on: ubuntu-latest
+    env:
+      RUSTC_WRAPPER: "sccache"
+      SCCACHE_GHA_ENABLED: "on"
+      PREVIEW_PATH: pr/${{ github.event.pull_request.number }}/docs
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: nightly-2024-05-02
+    - name: Install sccache
+      uses: mozilla-actions/sccache-action@v0.0.5
+
+    - name: Generate Docs
+      run: cargo doc --workspace --all-features --no-deps --document-private-items
```

Comment:
One open question I still have is: Should we use `--document-private-items`?
This is a good option for the `check_docs` job in the other workflow, but may not be good for the goals of this PR, as this PR is way more about a "simulation of what the actual docs would look like". (Which don't contain private items)

Reply:
Hum, yes.  Good point.  Let's remove `--document-private-items` as well

Reply:
Alright. I switched this setting off now, I think that's better. So no more `--document-private-items`.